File: 202. Happy Number.cpp
#include <unordered_map>
#include <cmath>
class Solution {
public:
    bool isHappy(int n) {
        unordered_map <int, int> dic;
        int sum = n;
        while (true)
        {
            string sumlist = to_string(sum);
            sum = 0;
            for (int i = 0; i < sumlist.length(); i++)
                sum += pow(sumlist[i] - '0', 2);
            if (sum == 1)
                return true;
            if (dic.count(sum) > 0)
                return false;
            dic[sum] = 1;
        }
    }
};

File: 202. Happy Number.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        dic = {}
        sum = n
        while True:
            sumlist = map(int, str(sum))
            sum = 0
            for i in sumlist:
                sum += i**2
            if sum == 1:
                return True
            if sum in dic:
                return False
            dic[sum] = 1
